interceptors/logging/payload.go

```diff
@@ -127,26 +147,14 @@ func PayloadStreamClientInterceptor(logger Logger, decider ClientPayloadLoggingD
 	return interceptors.StreamClientInterceptor(&payloadReportable{logger: logger, clientDecider: decider})
 }
 
-func logProtoMessageAsJson(logger Logger, pbMsg interface{}, key string, msg string) {
-	if p, ok := pbMsg.(proto.Message); ok {
-		payload, err := (&jsonpbObjectMarshaler{pb: p}).marshalJSON()
-		if err != nil {
-			logger = logger.With(key, err.Error())
-		} else {
-			logger = logger.With(key, string(payload))
-		}
-		logger.Log(INFO, msg)
-	}
-}
-
-type jsonpbObjectMarshaler struct {
-	pb proto.Message
-}
-
-func (j *jsonpbObjectMarshaler) marshalJSON() ([]byte, error) {
-	b := &bytes.Buffer{}
-	if err := JsonPbMarshaller.Marshal(b, j.pb); err != nil {
-		return nil, fmt.Errorf("jsonpb serializer failed: %v", err)
+func logProtoMessageAsJson(logger Logger, pbMsg proto.Message, key string, msg string) {
+	payload, err := protojson.Marshal(pbMsg)
+	if err != nil {
```

Comment:
So we can break this rule only for errors 😛 - https://www.bwplotka.dev/2020/how-thanos-would-program-in-go/#control-structure-prefer-early-returns-and-avoid-else

Reply:
No, we should break it only if it makes sense to do so.

I was thinking about this and since it's only oneliner it does not matter if we do with or without else. WDYT?

I think we could update style guide to mention for multi line bodies? 🤔